### ISSUE TITLE
shared memory definition

### ISSUE THREAD

**Issue by psychocoderHPC** [ComputationalRadiationPhysics/picongpu#518]
After every define of shared memory we must add `__syncthreads()`, if not we get a race condition warning from `cuda-memcheck --tool racecheck`

Please check if this is still the case with new cuda versions!

**Example: **
```C++
__shared__ int linearSuperCellIds[cellsInSuperCell];
__shared__ int hdf5ParticleOffset;

__syncthreads(); /*wait that all shared memory is initialized*/
```

**Comment by psychocoderHPC** [comment]
I removed all `__syncthreads()` after shared memory in our `dev` and checked with `cuda-memcheck --tool racecheck`
There are no race condition without the syncthreads in cuda 7.0

A pull request is on the way.

**Comment by ax3l** [comment]
is there an *official* documentation that usage of `__shared__` is save after declaration without a `__syncthreads()` in between?

Or is the `cuda-memcheck` in cuda 7.0 just lazier? (detection patterns changed already in the past.) I would actually prefer to have a trustworthy reference before changing it, just because the tool says "it's currently save"/"could not find a problem".

**Comment by psychocoderHPC** [comment]
In the official examples there are never a `__synchronize` after shared memory declaration. There is also no reason to put it in. This was added in PIConGPU by you and me based on a warning in cuda 5.0 or 5.5
If cuda 6.5 also creates no warning I will push a pull request.

**Comment by ax3l** [comment]
ok official examples are fine, too. thx for checking them!

**Comment by ax3l** [comment]
closed with #1082 and #1083
